core/src/main/java/com/yahoo/imapnio/async/request/ExtendedSearchSequence.java

```diff
@@ -0,0 +1,81 @@
+package com.yahoo.imapnio.async.request;
+
+import java.io.IOException;
+
+import javax.annotation.Nonnull;
+import javax.mail.search.SearchException;
+import javax.mail.search.SearchTerm;
+
+import com.sun.mail.iap.Argument;
+import com.sun.mail.imap.protocol.SearchSequence;
+import com.yahoo.imapnio.async.data.ExtendedModifiedSinceTerm;
+import com.yahoo.imapnio.async.exception.ImapAsyncClientException;
+
+/**
+ * This class extends the search sequence for modification sequence with the optional fields entry
+ * name and type defined in https://tools.ietf.org/html/rfc7162#section-3.1.5.
+ */
+public class ExtendedSearchSequence extends SearchSequence {
+
+    /** Literal for MODSEQ. */
+    private static final String MODSEQ = "MODSEQ";
+
+    /**
+     * Generate the IMAP search sequence for the given search expression.
+     *
+     * @param term the search term
+     * @return the IMAP search sequence argument
+     * @throws SearchException will not throw
```

Comment:
Let's not leave  the description as `will not throw`. Find the reason why this excpetion may be thrown and add it here.
Do the same for all @throws